### PR TITLE
ES6 features

### PR DESCRIPTION
**Pull request by NiedziolkaMichal**
This PR replaces old syntax with ES6 features as requested in #881 

- `const` is used for variable declarations. Only in cases of reassignment `let` was used instead.
- Only arrow functions are now passed as a callback.
- async/await is used instead of then/catch/finally (only 1 case in editable-wat.js).
- Removed few unused imports.
- Replaced == with === in mixedHTML() & register()
- Reordered few declarations.
- Removed unused function showCustomExampleHTML()

### CODE REVIEW

File: editor/js/editable-wat.js [hunk 1]
```diff
@@ -8,51 +8,48 @@ import "../css/editor-libs/ui-fonts.css";
...
+import {getEditorContent, initCodeEditor, languageJavaScript, languageWAST} from "./editor-libs/codemirror-editor.js";
+
+(async function () {
+  const watCodeBlock = document.getElementById("static-wat");
+  const jsCodeBlock = document.getElementById("static-js");
+  const exampleFeature = watCodeBlock.dataset["feature"];
+  const execute = document.getElementById("execute");
+  const output = document.querySelector("#console code");
+  const reset = document.getElementById("reset");
+  const wabtInitialized = wabtConstructor();
```
Comment (queengooborg) [inline editor/js/editable-wat.js]: Now that we're using ESM and defining an async function, we no longer need a separate variable to handle the initialization of `wabt`.

```suggestion
  const wabt = await wabtConstructor();
```

File: editor/js/editable-wat.js [hunk 2]
```diff
@@ -8,51 +8,48 @@ import "../css/editor-libs/ui-fonts.css";
...
+
+  const tabContainer = document.getElementById("tab-container");
+  const tabs = tabContainer.querySelectorAll("button[role='tab']");
+  const tabList = document.getElementById("tablist");
+
+  let watCodeMirror;
+  let jsCodeMirror;
+  let liveContainer = "";
+  let staticContainer;
+  let wabt = await wabtInitialized;
```
Comment (queengooborg) [inline editor/js/editable-wat.js]:
```suggestion
```

File: editor/js/editor-libs/codemirror-editor.js [hunk 3]
```diff
@@ -3,19 +3,18 @@ import { EditorState } from "@codemirror/state";
 import * as commands from "@codemirror/commands";
```
Comment (queengooborg) [inline editor/js/editor-libs/codemirror-editor.js]: There's a lot of changes going on in this file that don't seem to be related to ES6 conversion, including formatting changes that don't match MDN's style.  Can we revert this file and defer any changes to another PR?

File: package.json [hunk 4]
```diff
@@ -105,7 +105,6 @@
     "@codemirror/autocomplete": "^6.3.0",
     "@codemirror/commands": "^6.1.1",
     "@codemirror/lang-css": "^6.0.0",
-    "@codemirror/lang-html": "^6.1.2",
```
Comment (queengooborg) [inline package.json]: Let's also revert this while we revert `codemirror-editor.js`.

**Review by queengooborg — CHANGES_REQUESTED** [review]

**Comment by queengooborg** [comment]
(Oh whoops, apologies for pushing a commit, I didn't realize I had write access... 🙃 )

**Review by queengooborg — APPROVED** [review]
I went ahead and applied the changes to speed up the merging of this PR -- thank you for your hard work in doing this conversion, let's merge it!

**Comment by NiedziolkaMichal** [comment]
That's completely ok.
Those changes to codemirror-editor.js weren't really important, so we can just skip them.

Thanks for review

**Comment by queengooborg** [comment]
I'm down to review those changes if you'd like to create a separate PR for them!
